apps/roam/src/utils/conceptConversion.ts

```diff
@@ -0,0 +1,180 @@
+import { DiscourseNode } from "./getDiscourseNodes";
+import getDiscourseRelations from "./getDiscourseRelations";
+import type { DiscourseRelation } from "./getDiscourseRelations";
+import { Database } from "@repo/database/types.gen";
+import type { SupabaseContext } from "~/utils/supabaseContext";
+
+type roamAlphaAPI = Window["roamAlphaAPI"];
+
+// When it's merged: import type { LocalConceptDataInput } from '@repo/dataase/input_types';
+type LocalConceptDataInput = Partial<Database["public"]["CompositeTypes"]["concept_local_input"]>;
+
+const getNodeExtraData = (node_uid: string, api: roamAlphaAPI): { author_uid: string, created: string, last_modified: string, page_uid: string } => {
+  const result = api.q(`[
+    :find
+      ?author_uid
+      ?page_uid
+      ?created
+      ?last_modified
+      :in $ ?block_uid
+      :where
+      [?block :block/uid ?block_uid]
+      [?block :create/user ?author_id]
+      [?author_id :user/uid ?author_uid]
+      [?block :create/time ?created]
+      [?block :edit/time ?last_modified]
+      [(get-else $ ?block :block/page ?block) ?page_id]
+      [?page_id :block/uid ?page_uid]
+  ]`, node_uid);
+  if (result.length != 1 || result[0].length != 4)
+    throw new Error("Invalid result from Roam query");
+
+  const [author_uid, page_uid, created_t, last_modified_t] = result[0] as [string, string, number, number];
+  const created = new Date(created_t).toISOString();
+  const last_modified = new Date(last_modified_t).toISOString();
+  return {
+    author_uid,
+    created,
+    last_modified,
+    page_uid
+  }
+}
+
+
+export function discourseNodeSchemaToLocalConcept(
+  context: SupabaseContext,
+  node: DiscourseNode, // contains type, uid, text
+): LocalConceptDataInput {
+  const extra_data = getNodeExtraData(node.type, context.api);
+  const title_parts = node.text.split('/');
+  return {
+    space_id: context.spaceId,
+    author_local_id: extra_data.author_uid,
+    created: extra_data.created,
+    last_modified: extra_data.last_modified,
+    name: title_parts[title_parts.length - 1],
+    represented_by_local_id: node.type,
+    is_schema: true,
+  };
+}
+
+
+export function discourseNodeBlockToLocalConcept(
```

Comment:
Please review for Style Guide adherence (arrow function, named params, variable names, etc)

Reply:
Was this updated?  GitHub is still showing `function`